core/src/main/java/org/jivesoftware/sparkimpl/certificates/SparkTrustManager.java

```diff
@@ -187,6 +188,17 @@ public void checkServerTrusted(X509Certificate[] chain, String authType) throws
     }
 
     /**
+     * Return true if the certificate chain contain only one Self Signed certificate
+     */
+    private boolean isSelfSigned(X509Certificate[] chain) {
+        if(chain[0].getIssuerX500Principal().getName().equals(chain[0].getSubjectX500Principal().getName())){
```

Comment:
Perhaps replace by: `return chain[0].getIssuerX500Principal().getName().equals(chain[0].getSubjectX500Principal().getName())` (there's no need for the if/else)?